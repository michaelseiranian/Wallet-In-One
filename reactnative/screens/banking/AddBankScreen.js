import { StatusBar } from 'expo-status-bar';
import { useEffect, useState } from 'react'
import { StyleSheet, Text, View, ScrollView, Button, FlatList, TouchableOpacity, Image, TextInput,ActivityIndicator, Alert} from 'react-native';
import AuthWebView from './AuthView';
import { auth_get, auth_post} from '../../authentication'
import Loading from './Loading'
import { useTheme } from 'reactnative/src/theme/ThemeProvider'
import {styles} from 'reactnative/screens/All_Styles.style.js'

export default function AddBankScreen({ navigation }) {
    const [ search, setSearch ] = useState('') // Stores contents of search box
    const [ data, setData ] = useState([])     // Stores all bank data
    const [ bankData, setbankData ] = useState([])        // Stores filtered bankdata
    const [ isLoading, setIsLoading ] = useState(true)    // Initial load

    const [ bankAuthURL, setbankAuthURL ] = useState(null)// Link to authenticate with selected bank
    const [ authComplete, setAuthComplete ] = useState(false)
    const [ savedBanks, setSavedBanks ] = useState(null)

    const {dark, colors, setScheme} = useTheme();

    const reset = () => {
        setbankAuthURL(null)
        setAuthComplete(false)
        setSavedBanks(null)
    }

    const stylesInternal = StyleSheet.create({
        bankingContainer: {
            width: '100%',
            paddingLeft: 20,
            paddingRight: 20,
            paddingBottom: 0,
            borderWidth: 1,
            borderRadius: 5,
            borderColor: dark ? colors.background : '#ddd',
            overflow: 'hidden',
            backgroundColor: colors.background
          },
          bankingItem:{
            flexDirection: 'row',
            alignItems: 'center',
            padding: 10
          },
          bankingImage:{
              width: 50, 
              height: 50,
              marginRight: 10,
              resizeMode: 'contain',
          },
          bankingInput:{
            height: 40,
            width: '100%',
            borderWidth: 0.5,
            padding: 10,
            borderColor: 'gray',
            borderRadius: 5,
            marginTop: 5,
            marginBottom: 5,
            color: colors.text,
            backgroundColor: colors.background
          },
      });
    
    
    useEffect(() =>{
        const fetchData = async () => {
            console.log('fetching')
            const response = await auth_get('/banking/bank_list/')
            if (response.status == 200){
                setData(response.body)
                setbankData(response.body)
                setIsLoading(false)
            }
        }
        fetchData()
    }, [])
    
    const selectItem = (item) => {

        const getAuthURL = async (id) => {
            console.log('fetching url')
            const response = await auth_get(`/banking/auth_page/${id}/`)
            if (response.status == 200){
                setbankAuthURL(response.body.url)
            }
            else{
                console.log('error', response)
            }
        }
        console.log(item.id)
        getAuthURL(item.id)
    }

    const updateSearch = (string) => {
        setSearch(string)
        const searchString = string.toLowerCase()
        setbankData(data.filter(item => item.name.toLowerCase().includes(searchString)))
    }

    const detectFinish = (event) => {
        if (!authComplete && event.url.includes('example.com')) {
            setAuthComplete(true)
            updateServer(bankAuthURL)
            setbankAuthURL(null)
        }
    }

    const updateServer = (url) => {
        const sendLink = async () => {
            const response = await auth_post('/banking/finish_auth/',{'url': url})
            console.log(response)
            if (response.status == 200){
                setSavedBanks(response.body)
            }
            else if (response.status == 400 && response.body.error){
                Alert.alert("Error", response.body.error)
                reset()
            }
            else{
                Alert.alert("Error", "There was an error with the server, try again")
                reset()
            }
        }
        sendLink()
    }

    if (isLoading){
        return <Loading/>
    }

    if (bankAuthURL){
        return <AuthWebView url={bankAuthURL} onCancel={()=>{setbankAuthURL(null)}} stateChange={detectFinish} />
    }

    if (authComplete){
        return (
            <View
                style={styles(dark, colors).container}
            >
                {!savedBanks ? (
                    <TouchableOpacity style={{flex: 1,justifyContent:'center', alignItems:'center'}} onPress={()=>setAuthComplete(false)}>
                        <Text style={styles(dark, colors).text}>Bank Authentication Finished</Text>
                        <Text style={styles(dark, colors).text}>Waiting For server</Text>
                        <ActivityIndicator/>
                    </TouchableOpacity>
                ):(
                    <>
                        <View style={{flex: 1,justifyContent:'center', alignItems:'center'}}>
                            <Text style={styles(dark, colors).text}> Bank account(s) have been added</Text>
                        </View>
                        <Button title="Back" onPress={reset}/>
                    </>
                )}
            </View>
        )
    }
    
    return (
        <View
            style={[{flex:1, margin: 4, marginBottom: 54}]}
        >
                <TextInput
                    autoCapitalize='none'
                    autoCorrect={false}
                    style={stylesInternal.bankingInput}
                    placeholder='Search'
                    placeholderTextColor= {colors.text}
                    value={search}
                    onChangeText={updateSearch}
                />
                <View style={stylesInternal.bankingContainer}>
                    <FlatList data={bankData} renderItem={({item, index}) =>{
                        return (
                            <TouchableOpacity onPress={()=>selectItem(item)} style={stylesInternal.bankingItem}>
                                <Image                                   
                                    source={{ uri: item.logo }}
                                    style={stylesInternal.bankingImage}
                                />
                                <Text style={styles(dark, colors).text} key={index}>{item.name}</Text>
                            </TouchableOpacity>)
                        }}
                        ListEmptyComponent={<Text style={styles(dark, colors).text}>{'\nNo banks found\n'}</Text>}
                    />
                </View>
        </View>
    );
}
