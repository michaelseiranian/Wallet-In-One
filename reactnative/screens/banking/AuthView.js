import React, { useState } from 'react';
import { ActivityIndicator, View, Button } from 'react-native';
import { WebView } from 'react-native-webview'
import Loading from './Loading'
import { useTheme } from 'reactnative/src/theme/ThemeProvider'
import {styles} from 'reactnative/screens/All_Styles.style.js'

export default function AuthWebView({ url, onCancel, stateChange}) {
  const [isLoading, setIsLoading] = useState(true);
  const {dark, colors, setScheme} = useTheme();

  return (
    <View
    style={styles(dark, colors).container}
    >
      {isLoading ? (
        <Loading/>
      ) : null}
      <Button title="Cancel" onPress={onCancel}/>
      <WebView
        source={{ uri: url }}
        onLoad={() => setIsLoading(false)}
        onNavigationStateChange={stateChange}
      />
    </View>
  );
};