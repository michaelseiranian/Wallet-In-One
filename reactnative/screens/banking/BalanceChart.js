import { VictoryChart, VictoryLine, VictoryAxis, VictoryScatter } from "victory-native";	
import { useTheme } from 'reactnative/src/theme/ThemeProvider'
export function BalanceChart({rawData,highest}) {	

  const {dark, colors, setScheme} = useTheme();

  const data = Object.keys(rawData).map(date => ({	
    x: new Date(date),	
    y: parseFloat(rawData[date])	
  }));	

  const beforeMonths = Object.keys(rawData).map(date => {date = new Date(date); return new Date(date.getFullYear(), date.getMonth(), 1)});	
  const afterMonths = Object.keys(rawData).map(date => {date = new Date(date); return new Date(date.getFullYear(), date.getMonth()+1, 1)});	
  const months = beforeMonths.concat(afterMonths)	
  	
  return (	
    <VictoryChart minDomain={{ y: 0 }} maxDomain={{ y: highest*1.1 }} height={220} width={350}>	
    <VictoryLine data={data} interpolation="stepAfter" style={{ data: { stroke: "#0055b3", strokeWidth: 2 } }} />	
    <VictoryAxis	
      tickFormat={(date) => {	
        var d = new Date(date)	
        return `${d.getMonth()+1}/${d.getFullYear().toString().substr(-2)}`	
      }}	
      tickValues = {months}	
      style={{	
        grid: {	
          stroke: "grey",	
          strokeDsharray: "2, 5"	
        },	
        axis: {stroke: 'grey'},
        tickLabels: {fill: colors.text , fontSize: 12},
      }}	
    />	
    <VictoryAxis	
      tickFormat={(value) => `Â£${value}`}	
      dependentAxis={true}

      style={{	
        grid: {	
          stroke: "grey",	
          strokeDasharray: "2, 5"	
        },
        axis: {stroke: 'grey'},
        tickLabels: {fill: colors.text},	
      }}	
    />	
    	
  </VictoryChart>	
  );	
}