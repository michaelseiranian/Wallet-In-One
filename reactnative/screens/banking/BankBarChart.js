import { VictoryChart, VictoryBar, VictoryAxis, VictoryScatter } from "victory-native";	
import { Text, Dimensions } from "react-native";
import { useTheme } from 'reactnative/src/theme/ThemeProvider'
import {styles} from 'reactnative/screens/All_Styles.style.js'

import { BarChart } from "react-native-chart-kit";
export function BankBarChart({rawData, tab}) {

  const {dark, colors, setScheme} = useTheme();
  
  const data = {
    labels: rawData.labels,
    datasets: [
      {
        data: rawData.values
      }
    ],
  };

  const chartConfig = {
    backgroundColor: colors.background,
    backgroundGradientFrom: colors.background,
    backgroundGradientFromOpacity: 0,
    backgroundGradientTo: colors.background,
    backgroundGradientToOpacity: 0.5,
    color: (opacity) => {
      if (tab == 1){
        return `rgba(26, 181, 9, ${opacity})`
      }
      if (tab == 2){
        return `rgba(300,0,0, ${opacity})`
      }
      else{
        return `rgba(6, 146, 207, ${opacity})`
      }
    },
    labelColor: (opacity = 1) => colors.text,
    strokeWidth: 1, // optional, default 3
    barPercentage: 0.5, 
    useShadowColorFromDataset: false // optional
  };

  return (
    <BarChart
      data={data}
      width={Dimensions.get("window").width*0.8}
      height={200}
      yAxisLabel="Â£"
      fromZero={true}
      chartConfig={chartConfig}
      style={{}}
    />
  )
}