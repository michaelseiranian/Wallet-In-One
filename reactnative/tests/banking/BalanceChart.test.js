import React from 'react';
import renderer from 'react-test-renderer';
import { BalanceChart } from '../../screens/banking/BalanceChart'

describe('<BalanceChart />', () => {
    it('BalanceChart test', () => {
        var balance_history = {
            "2023-02-15": 100.0,
            "2023-02-01": 150.0,
            "2023-01-15": 50.0,
            "2023-01-01": 100.0,
            "2022-12-15": -200.0,
            "2022-12-01": -150.0
        }

        const snapshot = renderer.create(<BalanceChart rawData={balance_history} highest={150.0}/>);
        expect(snapshot).toMatchSnapshot();         
      });
  });
