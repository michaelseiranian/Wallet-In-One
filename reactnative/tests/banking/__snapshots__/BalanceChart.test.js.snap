// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BalanceChart /> BalanceChart test 1`] = `
<View
  className="VictoryContainer"
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  pointerEvents="box-none"
  style={
    {
      "position": "relative",
    }
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="220"
    bbWidth="350"
    focusable={false}
    height={220}
    meetOrSlice={0}
    minX={0}
    minY={0}
    role="img"
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "height": "100%",
          "width": "100%",
        },
        {
          "flex": 0,
          "height": 220,
          "width": 350,
        },
      ]
    }
    vbHeight={220}
    vbWidth={350}
    width={350}
  >
    <RNSVGGroup>
      <RNSVGRect
        fill={null}
        height="220"
        propList={
          [
            "fill",
          ]
        }
        width="350"
        x="0"
        y="0"
      />
      <RNSVGGroup
        clipPath="victory-clip-2"
      >
        <RNSVGGroup>
          <RNSVGDefs>
            <RNSVGClipPath
              name="victory-clip-2"
            >
              <RNSVGRect
                height="120"
                vectorEffect={1}
                width="250"
                x="50"
                y="50"
              />
            </RNSVGClipPath>
          </RNSVGDefs>
        </RNSVGGroup>
        <RNSVGPath
          d="M50,279.0909090909091L88.88888888888889,279.0909090909091L88.88888888888889,315.4545454545454L136.11111111111111,315.4545454545454L136.11111111111111,97.27272727272728L175,97.27272727272728L175,133.63636363636365L222.2222222222222,133.63636363636365L222.2222222222222,60.909090909090914L261.11111111111114,60.909090909090914L261.11111111111114,97.27272727272728"
          fill={
            {
              "payload": 0,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
            ]
          }
          stroke={
            {
              "payload": 4278212019,
              "type": 0,
            }
          }
          strokeWidth="2"
        />
      </RNSVGGroup>
      <RNSVGGroup>
        <RNSVGLine
          fill={
            {
              "payload": 0,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4286611584,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="1"
          vectorEffect={1}
          x1="50"
          x2="300"
          y1="170"
          y2="170"
        />
        <RNSVGGroup>
          <RNSVGLine
            fill={null}
            propList={
              [
                "fill",
                "stroke",
              ]
            }
            stroke={
              {
                "payload": 4286611584,
                "type": 0,
              }
            }
            vectorEffect={1}
            x1="222.2222222222222"
            x2="222.2222222222222"
            y1="170"
            y2="50"
          />
          <RNSVGText
            content={null}
            dx={
              [
                "0",
              ]
            }
            dy={[]}
            font={{}}
            name="chart-axis-1-tickLabels-0"
            rotate={[]}
            x={
              [
                "222.2222222222222",
              ]
            }
            y={
              [
                "191.26",
              ]
            }
          >
            <RNSVGTSpan
              content="2/23"
              dx={
                [
                  "0",
                ]
              }
              dy={
                [
                  "0",
                ]
              }
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              font={
                {
                  "fontFamily": "Gill Sans",
                  "fontSize": "12",
                  "letterSpacing": "normal",
                  "textAnchor": "middle",
                }
              }
              propList={
                [
                  "fill",
                ]
              }
              rotate={[]}
              x={
                [
                  "222.2222222222222",
                ]
              }
              y={[]}
            />
          </RNSVGText>
        </RNSVGGroup>
        <RNSVGGroup>
          <RNSVGLine
            fill={null}
            propList={
              [
                "fill",
                "stroke",
              ]
            }
            stroke={
              {
                "payload": 4286611584,
                "type": 0,
              }
            }
            vectorEffect={1}
            x1="222.2222222222222"
            x2="222.2222222222222"
            y1="170"
            y2="50"
          />
          <RNSVGText
            content={null}
            dx={
              [
                "0",
              ]
            }
            dy={[]}
            font={{}}
            name="chart-axis-1-tickLabels-1"
            rotate={[]}
            x={
              [
                "222.2222222222222",
              ]
            }
            y={
              [
                "191.26",
              ]
            }
          >
            <RNSVGTSpan
              content="2/23"
              dx={
                [
                  "0",
                ]
              }
              dy={
                [
                  "0",
                ]
              }
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              font={
                {
                  "fontFamily": "Gill Sans",
                  "fontSize": "12",
                  "letterSpacing": "normal",
                  "textAnchor": "middle",
                }
              }
              propList={
                [
                  "fill",
                ]
              }
              rotate={[]}
              x={
                [
                  "222.2222222222222",
                ]
              }
              y={[]}
            />
          </RNSVGText>
        </RNSVGGroup>
        <RNSVGGroup>
          <RNSVGLine
            fill={null}
            propList={
              [
                "fill",
                "stroke",
              ]
            }
            stroke={
              {
                "payload": 4286611584,
                "type": 0,
              }
            }
            vectorEffect={1}
            x1="136.11111111111111"
            x2="136.11111111111111"
            y1="170"
            y2="50"
          />
          <RNSVGText
            content={null}
            dx={
              [
                "0",
              ]
            }
            dy={[]}
            font={{}}
            name="chart-axis-1-tickLabels-2"
            rotate={[]}
            x={
              [
                "136.11111111111111",
              ]
            }
            y={
              [
                "191.26",
              ]
            }
          >
            <RNSVGTSpan
              content="1/23"
              dx={
                [
                  "0",
                ]
              }
              dy={
                [
                  "0",
                ]
              }
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              font={
                {
                  "fontFamily": "Gill Sans",
                  "fontSize": "12",
                  "letterSpacing": "normal",
                  "textAnchor": "middle",
                }
              }
              propList={
                [
                  "fill",
                ]
              }
              rotate={[]}
              x={
                [
                  "136.11111111111111",
                ]
              }
              y={[]}
            />
          </RNSVGText>
        </RNSVGGroup>
        <RNSVGGroup>
          <RNSVGLine
            fill={null}
            propList={
              [
                "fill",
                "stroke",
              ]
            }
            stroke={
              {
                "payload": 4286611584,
                "type": 0,
              }
            }
            vectorEffect={1}
            x1="136.11111111111111"
            x2="136.11111111111111"
            y1="170"
            y2="50"
          />
          <RNSVGText
            content={null}
            dx={
              [
                "0",
              ]
            }
            dy={[]}
            font={{}}
            name="chart-axis-1-tickLabels-3"
            rotate={[]}
            x={
              [
                "136.11111111111111",
              ]
            }
            y={
              [
                "191.26",
              ]
            }
          >
            <RNSVGTSpan
              content="1/23"
              dx={
                [
                  "0",
                ]
              }
              dy={
                [
                  "0",
                ]
              }
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              font={
                {
                  "fontFamily": "Gill Sans",
                  "fontSize": "12",
                  "letterSpacing": "normal",
                  "textAnchor": "middle",
                }
              }
              propList={
                [
                  "fill",
                ]
              }
              rotate={[]}
              x={
                [
                  "136.11111111111111",
                ]
              }
              y={[]}
            />
          </RNSVGText>
        </RNSVGGroup>
        <RNSVGGroup>
          <RNSVGLine
            fill={null}
            propList={
              [
                "fill",
                "stroke",
              ]
            }
            stroke={
              {
                "payload": 4286611584,
                "type": 0,
              }
            }
            vectorEffect={1}
            x1="50"
            x2="50"
            y1="170"
            y2="50"
          />
          <RNSVGText
            content={null}
            dx={
              [
                "0",
              ]
            }
            dy={[]}
            font={{}}
            name="chart-axis-1-tickLabels-4"
            rotate={[]}
            x={
              [
                "50",
              ]
            }
            y={
              [
                "191.26",
              ]
            }
          >
            <RNSVGTSpan
              content="12/22"
              dx={
                [
                  "0",
                ]
              }
              dy={
                [
                  "0",
                ]
              }
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              font={
                {
                  "fontFamily": "Gill Sans",
                  "fontSize": "12",
                  "letterSpacing": "normal",
                  "textAnchor": "middle",
                }
              }
              propList={
                [
                  "fill",
                ]
              }
              rotate={[]}
              x={
                [
                  "50",
                ]
              }
              y={[]}
            />
          </RNSVGText>
        </RNSVGGroup>
        <RNSVGGroup>
          <RNSVGLine
            fill={null}
            propList={
              [
                "fill",
                "stroke",
              ]
            }
            stroke={
              {
                "payload": 4286611584,
                "type": 0,
              }
            }
            vectorEffect={1}
            x1="50"
            x2="50"
            y1="170"
            y2="50"
          />
          <RNSVGText
            content={null}
            dx={
              [
                "0",
              ]
            }
            dy={[]}
            font={{}}
            name="chart-axis-1-tickLabels-5"
            rotate={[]}
            x={
              [
                "50",
              ]
            }
            y={
              [
                "191.26",
              ]
            }
          >
            <RNSVGTSpan
              content="12/22"
              dx={
                [
                  "0",
                ]
              }
              dy={
                [
                  "0",
                ]
              }
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              font={
                {
                  "fontFamily": "Gill Sans",
                  "fontSize": "12",
                  "letterSpacing": "normal",
                  "textAnchor": "middle",
                }
              }
              propList={
                [
                  "fill",
                ]
              }
              rotate={[]}
              x={
                [
                  "50",
                ]
              }
              y={[]}
            />
          </RNSVGText>
        </RNSVGGroup>
        <RNSVGGroup>
          <RNSVGLine
            fill={null}
            propList={
              [
                "fill",
                "stroke",
              ]
            }
            stroke={
              {
                "payload": 4286611584,
                "type": 0,
              }
            }
            vectorEffect={1}
            x1="300"
            x2="300"
            y1="170"
            y2="50"
          />
          <RNSVGText
            content={null}
            dx={
              [
                "0",
              ]
            }
            dy={[]}
            font={{}}
            name="chart-axis-1-tickLabels-6"
            rotate={[]}
            x={
              [
                "300",
              ]
            }
            y={
              [
                "191.26",
              ]
            }
          >
            <RNSVGTSpan
              content="3/23"
              dx={
                [
                  "0",
                ]
              }
              dy={
                [
                  "0",
                ]
              }
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              font={
                {
                  "fontFamily": "Gill Sans",
                  "fontSize": "12",
                  "letterSpacing": "normal",
                  "textAnchor": "middle",
                }
              }
              propList={
                [
                  "fill",
                ]
              }
              rotate={[]}
              x={
                [
                  "300",
                ]
              }
              y={[]}
            />
          </RNSVGText>
        </RNSVGGroup>
        <RNSVGGroup>
          <RNSVGLine
            fill={null}
            propList={
              [
                "fill",
                "stroke",
              ]
            }
            stroke={
              {
                "payload": 4286611584,
                "type": 0,
              }
            }
            vectorEffect={1}
            x1="300"
            x2="300"
            y1="170"
            y2="50"
          />
          <RNSVGText
            content={null}
            dx={
              [
                "0",
              ]
            }
            dy={[]}
            font={{}}
            name="chart-axis-1-tickLabels-7"
            rotate={[]}
            x={
              [
                "300",
              ]
            }
            y={
              [
                "191.26",
              ]
            }
          >
            <RNSVGTSpan
              content="3/23"
              dx={
                [
                  "0",
                ]
              }
              dy={
                [
                  "0",
                ]
              }
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              font={
                {
                  "fontFamily": "Gill Sans",
                  "fontSize": "12",
                  "letterSpacing": "normal",
                  "textAnchor": "middle",
                }
              }
              propList={
                [
                  "fill",
                ]
              }
              rotate={[]}
              x={
                [
                  "300",
                ]
              }
              y={[]}
            />
          </RNSVGText>
        </RNSVGGroup>
        <RNSVGGroup>
          <RNSVGLine
            fill={null}
            propList={
              [
                "fill",
                "stroke",
              ]
            }
            stroke={
              {
                "payload": 4286611584,
                "type": 0,
              }
            }
            vectorEffect={1}
            x1="222.2222222222222"
            x2="222.2222222222222"
            y1="170"
            y2="50"
          />
          <RNSVGText
            content={null}
            dx={
              [
                "0",
              ]
            }
            dy={[]}
            font={{}}
            name="chart-axis-1-tickLabels-8"
            rotate={[]}
            x={
              [
                "222.2222222222222",
              ]
            }
            y={
              [
                "191.26",
              ]
            }
          >
            <RNSVGTSpan
              content="2/23"
              dx={
                [
                  "0",
                ]
              }
              dy={
                [
                  "0",
                ]
              }
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              font={
                {
                  "fontFamily": "Gill Sans",
                  "fontSize": "12",
                  "letterSpacing": "normal",
                  "textAnchor": "middle",
                }
              }
              propList={
                [
                  "fill",
                ]
              }
              rotate={[]}
              x={
                [
                  "222.2222222222222",
                ]
              }
              y={[]}
            />
          </RNSVGText>
        </RNSVGGroup>
        <RNSVGGroup>
          <RNSVGLine
            fill={null}
            propList={
              [
                "fill",
                "stroke",
              ]
            }
            stroke={
              {
                "payload": 4286611584,
                "type": 0,
              }
            }
            vectorEffect={1}
            x1="222.2222222222222"
            x2="222.2222222222222"
            y1="170"
            y2="50"
          />
          <RNSVGText
            content={null}
            dx={
              [
                "0",
              ]
            }
            dy={[]}
            font={{}}
            name="chart-axis-1-tickLabels-9"
            rotate={[]}
            x={
              [
                "222.2222222222222",
              ]
            }
            y={
              [
                "191.26",
              ]
            }
          >
            <RNSVGTSpan
              content="2/23"
              dx={
                [
                  "0",
                ]
              }
              dy={
                [
                  "0",
                ]
              }
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              font={
                {
                  "fontFamily": "Gill Sans",
                  "fontSize": "12",
                  "letterSpacing": "normal",
                  "textAnchor": "middle",
                }
              }
              propList={
                [
                  "fill",
                ]
              }
              rotate={[]}
              x={
                [
                  "222.2222222222222",
                ]
              }
              y={[]}
            />
          </RNSVGText>
        </RNSVGGroup>
        <RNSVGGroup>
          <RNSVGLine
            fill={null}
            propList={
              [
                "fill",
                "stroke",
              ]
            }
            stroke={
              {
                "payload": 4286611584,
                "type": 0,
              }
            }
            vectorEffect={1}
            x1="136.11111111111111"
            x2="136.11111111111111"
            y1="170"
            y2="50"
          />
          <RNSVGText
            content={null}
            dx={
              [
                "0",
              ]
            }
            dy={[]}
            font={{}}
            name="chart-axis-1-tickLabels-10"
            rotate={[]}
            x={
              [
                "136.11111111111111",
              ]
            }
            y={
              [
                "191.26",
              ]
            }
          >
            <RNSVGTSpan
              content="1/23"
              dx={
                [
                  "0",
                ]
              }
              dy={
                [
                  "0",
                ]
              }
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              font={
                {
                  "fontFamily": "Gill Sans",
                  "fontSize": "12",
                  "letterSpacing": "normal",
                  "textAnchor": "middle",
                }
              }
              propList={
                [
                  "fill",
                ]
              }
              rotate={[]}
              x={
                [
                  "136.11111111111111",
                ]
              }
              y={[]}
            />
          </RNSVGText>
        </RNSVGGroup>
        <RNSVGGroup>
          <RNSVGLine
            fill={null}
            propList={
              [
                "fill",
                "stroke",
              ]
            }
            stroke={
              {
                "payload": 4286611584,
                "type": 0,
              }
            }
            vectorEffect={1}
            x1="136.11111111111111"
            x2="136.11111111111111"
            y1="170"
            y2="50"
          />
          <RNSVGText
            content={null}
            dx={
              [
                "0",
              ]
            }
            dy={[]}
            font={{}}
            name="chart-axis-1-tickLabels-11"
            rotate={[]}
            x={
              [
                "136.11111111111111",
              ]
            }
            y={
              [
                "191.26",
              ]
            }
          >
            <RNSVGTSpan
              content="1/23"
              dx={
                [
                  "0",
                ]
              }
              dy={
                [
                  "0",
                ]
              }
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              font={
                {
                  "fontFamily": "Gill Sans",
                  "fontSize": "12",
                  "letterSpacing": "normal",
                  "textAnchor": "middle",
                }
              }
              propList={
                [
                  "fill",
                ]
              }
              rotate={[]}
              x={
                [
                  "136.11111111111111",
                ]
              }
              y={[]}
            />
          </RNSVGText>
        </RNSVGGroup>
      </RNSVGGroup>
      <RNSVGGroup>
        <RNSVGLine
          fill={
            {
              "payload": 0,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4286611584,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="1"
          vectorEffect={1}
          x1="50"
          x2="50"
          y1="50"
          y2="170"
        />
        <RNSVGGroup>
          <RNSVGLine
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeDasharray",
              ]
            }
            stroke={
              {
                "payload": 4286611584,
                "type": 0,
              }
            }
            strokeDasharray={
              [
                "2",
                "5",
              ]
            }
            vectorEffect={1}
            x1="50"
            x2="300"
            y1="133.63636363636365"
            y2="133.63636363636365"
          />
          <RNSVGText
            content={null}
            dx={
              [
                "0",
              ]
            }
            dy={[]}
            font={{}}
            name="chart-axis-2-tickLabels-0"
            rotate={[]}
            x={
              [
                "39",
              ]
            }
            y={
              [
                "138.60636363636365",
              ]
            }
          >
            <RNSVGTSpan
              content="£50"
              dx={
                [
                  "0",
                ]
              }
              dy={
                [
                  "0",
                ]
              }
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              font={
                {
                  "fontFamily": "Gill Sans",
                  "fontSize": "14",
                  "letterSpacing": "normal",
                  "textAnchor": "end",
                }
              }
              propList={
                [
                  "fill",
                ]
              }
              rotate={[]}
              x={
                [
                  "39",
                ]
              }
              y={[]}
            />
          </RNSVGText>
        </RNSVGGroup>
        <RNSVGGroup>
          <RNSVGLine
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeDasharray",
              ]
            }
            stroke={
              {
                "payload": 4286611584,
                "type": 0,
              }
            }
            strokeDasharray={
              [
                "2",
                "5",
              ]
            }
            vectorEffect={1}
            x1="50"
            x2="300"
            y1="97.27272727272728"
            y2="97.27272727272728"
          />
          <RNSVGText
            content={null}
            dx={
              [
                "0",
              ]
            }
            dy={[]}
            font={{}}
            name="chart-axis-2-tickLabels-1"
            rotate={[]}
            x={
              [
                "39",
              ]
            }
            y={
              [
                "102.24272727272728",
              ]
            }
          >
            <RNSVGTSpan
              content="£100"
              dx={
                [
                  "0",
                ]
              }
              dy={
                [
                  "0",
                ]
              }
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              font={
                {
                  "fontFamily": "Gill Sans",
                  "fontSize": "14",
                  "letterSpacing": "normal",
                  "textAnchor": "end",
                }
              }
              propList={
                [
                  "fill",
                ]
              }
              rotate={[]}
              x={
                [
                  "39",
                ]
              }
              y={[]}
            />
          </RNSVGText>
        </RNSVGGroup>
        <RNSVGGroup>
          <RNSVGLine
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeDasharray",
              ]
            }
            stroke={
              {
                "payload": 4286611584,
                "type": 0,
              }
            }
            strokeDasharray={
              [
                "2",
                "5",
              ]
            }
            vectorEffect={1}
            x1="50"
            x2="300"
            y1="60.909090909090914"
            y2="60.909090909090914"
          />
          <RNSVGText
            content={null}
            dx={
              [
                "0",
              ]
            }
            dy={[]}
            font={{}}
            name="chart-axis-2-tickLabels-2"
            rotate={[]}
            x={
              [
                "39",
              ]
            }
            y={
              [
                "65.87909090909092",
              ]
            }
          >
            <RNSVGTSpan
              content="£150"
              dx={
                [
                  "0",
                ]
              }
              dy={
                [
                  "0",
                ]
              }
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              font={
                {
                  "fontFamily": "Gill Sans",
                  "fontSize": "14",
                  "letterSpacing": "normal",
                  "textAnchor": "end",
                }
              }
              propList={
                [
                  "fill",
                ]
              }
              rotate={[]}
              x={
                [
                  "39",
                ]
              }
              y={[]}
            />
          </RNSVGText>
        </RNSVGGroup>
      </RNSVGGroup>
      <View
        pointerEvents="box-none"
        style={
          {
            "left": 0,
            "position": "absolute",
            "top": 0,
            "zIndex": 99,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="220"
          bbWidth="350"
          focusable={false}
          height={220}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "height": "100%",
                "overflow": "visible",
                "width": "100%",
              },
              {
                "flex": 0,
                "height": 220,
                "width": 350,
              },
            ]
          }
          vbHeight={220}
          vbWidth={350}
          width={350}
        >
          <RNSVGGroup />
        </RNSVGSvgView>
      </View>
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
